### PR TITLE
Integrate PalsMpiexecSettings into Experiment factory methods

### PR DESCRIPTION
**Pull request by MattToast**
Registers the `PalsMpiexecSettings` class in SmartsSim's `Experiment` factory methods so that the correct `_BaseMpiSettings` class is returned depending on the launcher the user is intending to use:
```py
>>> pbs_exp = Experiment(launcher="pbs", ...)
>>> rs_1 = pbs_exp.create_run_settings(run_command="mpiexec", ...)
>>> type(rs_1)  # returns a OpenMPI compliant RunSettings
<smartsim.settings.mpiSettings.MpiexecSettings object at 0x7f9fe02eee00>
>>> pals_exp = Experiment(launcher="pals", ...)
>>> rs_2 = pals_exp.create_run_settings(run_command="mpiexec", ...)
>>> type(rs_2)  # returns a PALS compliant RunSettings
<smartsim.settings.palsSettings.PalsMpiexecSettings object at 0x7f9fdaf6f1c0>
```

### CODE REVIEW

**Review by ashao — APPROVED** [review]
Thanks for getting this in!

File: smartsim/database/orchestrator.py [hunk 1]
```diff
@@ -64,6 +64,7 @@
 by_launcher: t.Dict[str, t.List[str]] = {
     "slurm": ["srun", "mpirun", "mpiexec"],
     "pbs": ["aprun", "mpirun", "mpiexec"],
+    "pals": ["mpiexec"],
```
[the remark this answers is not in the document]
Comment (al-rigazzi) [reply smartsim/database/orchestrator.py:67]: My suggestion is _not yet_, as I remember there were some issues. But we will have to in the future.

**Review by al-rigazzi — COMMENTED** [review]
LGTM, but I gotta ask if we ran the tests on any PALS machine because I'm the baaad guy

File: tests/test_configs/mpi_impl_stubs/pals/mpiexec [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+# DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+# SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+# OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+# Based on running mpiexec.slurm without loading a specific MPI
+# implementation 
+
+echo "mpiexec version 1.2.12 revision d3dd612f9372 built Apr 12 2023"
```
[the remark this answers is not in the document]
Comment (MattToast) [reply tests/test_configs/mpi_impl_stubs/pals/mpiexec:32]: Whoops... over eager copy and paste 🤦

File: smartsim/database/orchestrator.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (MattToast) [reply smartsim/database/orchestrator.py:67]: I think I'm inclined to agree as I didn't specifically test against `aprun`. I'll be sure to throw in a ticket to track the future work needed!

**Comment by MattToast** [comment]
@al-rigazzi I manually scaled down a couple of the `on_wlm` tests to run on an internal 2 node EX/PBS/PALS machine, and everything  seemed to be working as expected, but I was unable to run the test suite in full.

LMK if you think we should hold this off until we can do a more substantial test run!!

File: conftest.py [hunk 3]
```diff
@@ -144,19 +144,15 @@ def get_hostlist() -> t.Optional[t.List[str]]:
...
                 return None
         elif "PBS_NODEFILE" in os.environ and not shutil.which("aprun"):
             try:
                 pbs_fp = os.environ["PBS_NODEFILE"]
                 with open(pbs_fp, "r", encoding="utf-8") as nodefile:
                     lines = nodefile.readlines()
-                    test_hostlist = list(
-                        dict.fromkeys([line.strip() for line in lines])
-                    )
+                    test_hostlist = list({line.strip() for line in lines})
```
Comment (ankona) [inline conftest.py:155]: could we pull the 4 lines out into a function (144-147 and 152-155) so the process of de-duping and processing the node files is in one place?

Comment (MattToast) [reply conftest.py:155]: Good idea! I'll tack something on this PR rq!!

**Pull request merged**